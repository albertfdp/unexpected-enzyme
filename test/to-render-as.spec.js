import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React from 'react';
import PropTypes from 'prop-types';

const User = ({ children }) => (
  <div className="user">
    <h3 id="name">Harriet</h3>
    {children}
  </div>
);

User.propTypes = {
  children: PropTypes.node
};

describe('to-render-as', () => {
  describe('when it is a subtree', () => {
    it('passes when the actual matches the expected', () => {
      const wrapper = mount(
        <User>
          <div id="child">Children</div>
        </User>
      );

      expect(wrapper, 'to render as', <div id="child">Children</div>);
    });

    it('fails when the actual does not the expected', () => {
      const wrapper = mount(
        <User>
          <div id="child">Foo</div>
        </User>
      );

      expect(
        () => expect(wrapper, 'to render as', <div id="child">Children</div>),
        'with error matching snapshot'
      );
    });

    it('fails when the actual is not anywhere on the tree', () => {
      const wrapper = mount(
        <User>
          <div id="child">Foo</div>
        </User>
      );

      expect(
        () => expect(wrapper, 'to render as', <h3>Header</h3>),
        'with error matching snapshot'
      );
    });
  });
});
