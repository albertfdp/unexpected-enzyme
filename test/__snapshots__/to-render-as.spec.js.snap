// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-render-as when it is a subtree fails when the actual does not the expected 1`] = `
"expected
<User>
  <div className=\\"user\\">
    <h3 id=\\"name\\">
      Harriet
    </h3>
    <div id=\\"child\\">
      Foo
    </div>
  </div>
</User>
to render as <div id=\\"child\\">Children</div>

<User>
  <div id=\\"child\\">
    Foo // -Foo
        // +Children
  </div>
</User>"
`;

exports[`to-render-as when it is a subtree fails when the actual is not anywhere on the tree 1`] = `
"expected
<User>
  <div className=\\"user\\">
    <h3 id=\\"name\\">
      Harriet
    </h3>
    <div id=\\"child\\">
      Foo
    </div>
  </div>
</User>
to render as <h3>Header</h3>

<User // should be <h3
>
  <div id=\\"child\\">
    Foo // -Foo
        // +Header
  </div>
</User>"
`;
